raster-foundry/raster-foundry
# Prevent analysis creation for layers without AOIs defined

## Overview

This PR enforces (within the UI) that a layer has an AOI before running an analysis on it.

When attempting to create an analysis from:
* the layer scene list
* a scene's context menu
* the create-scene route (if someone somehow manages to get there with a layer that has no AOI

each scene will be checked for an AOI. If at least one scene does not have an AOI, a modal is presented to the user informing them of the need for an AOI and gives them a chance to define the AOI for the first layer without one.

### Checklist


## Testing Instructions

 * From the project layer list, select a few layers (some with AOIs, some without) and see that the flow makes sense. 

Closes #4753
